evmcore/dummy_block.go

```diff
@@ -45,6 +45,9 @@ type (
 		GasUsed  uint64
 
 		BaseFee *big.Int
+
+		Difficulty  *big.Int
```

Comment:
Since `Difficulty` is constant in Fantom (and presumably also in Sonic), I think this field is not required in our internal datastructures.